### PR TITLE
Credentials resolver returns `RemoteClusterCredentials`

### PR DESCRIPTION
**Pull request by n1v0lg**
A small refactor around resolving remote cluster authorization: rename `RemoteAccessCredentials` to `RemoteClusterCredentials` since the latter is more specific and descriptive, return `RemoteClusterCredentials` from the resolver, and finally, rename the relevant header to `REMOTE_CLUSTER_AUTHORIZATION_HEADER_KEY`. This is more precise and avoids the unnecessary use or "remote access". I will follow up with a bigger rename PR, replacing the terms "remote access" with "remote cluster" more broadly.

### CODE REVIEW

File: x-pack/plugin/security/qa/security-trial/src/javaRestTest/java/org/elasticsearch/xpack/security/remoteaccess/RemoteAccessHeadersForCcsRestIT.java [hunk 1]
```diff
@@ -393,9 +393,9 @@ private void expectActionsAndHeadersForCluster(
     }
 
     private void assertContainsRemoteClusterCredential(String clusterCredential, CapturedActionWithHeaders actual) {
-        assertThat(actual.headers(), hasKey(RemoteAccessHeaders.REMOTE_ACCESS_CLUSTER_CREDENTIAL_HEADER_KEY));
+        assertThat(actual.headers(), hasKey(RemoteAccessHeaders.REMOTE_CLUSTER_AUTHORIZATION_HEADER_KEY));
```
Comment (n1v0lg) [inline x-pack/plugin/security/qa/security-trial/src/javaRestTest/java/org/elasticsearch/xpack/security/remoteaccess/RemoteAccessHeadersForCcsRestIT.java:396]: "Authorization" aligns better with the setting name, and also the fact that the value itself is `ApiKey [encoded]` so essentially an authorization header value, so `REMOTE_CLUSTER_AUTHORIZATION_HEADER_KEY` feels more natural.

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java [hunk 2]
```diff
@@ -905,7 +905,7 @@ Collection<Object> createComponents(
         ipFilter.set(new IPFilter(settings, auditTrailService, clusterService.getClusterSettings(), getLicenseState()));
         components.add(ipFilter.get());
 
-        final RemoteClusterAuthorizationResolver remoteClusterAuthorizationResolver = new RemoteClusterAuthorizationResolver(
+        final RemoteClusterCredentialsResolver remoteClusterCredentialsResolver = new RemoteClusterCredentialsResolver(
```
Comment (n1v0lg) [inline x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java:908]: `RemoteClusterCredentialsResolver` now returns an instance of `RemoteClusterCredentials` which contains both the cluster alias and authorization value, hence the rename here.

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/RemoteAccessHeaders.java [hunk 3]
```diff
@@ -15,42 +15,42 @@
 
 public final class RemoteAccessHeaders {
 
-    public static final String REMOTE_ACCESS_CLUSTER_CREDENTIAL_HEADER_KEY = "_remote_access_cluster_credential";
-    private final String clusterCredentialsHeader;
+    public static final String REMOTE_CLUSTER_AUTHORIZATION_HEADER_KEY = "_remote_cluster_authorization";
```
Comment (n1v0lg) [inline x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/RemoteAccessHeaders.java:18]: This is a "breaking change" but we're behind a feature flag so I don't see a reason not make it.

**Review by ywangd — APPROVED** [review]
LGTM

**Comment by n1v0lg** [comment]
@elasticmachine update branch

**Pull request merged**